cmd/mqtt/main.go

```diff
@@ -122,6 +126,30 @@ func main() {
 		log.Fatalf(err.Error())
 	}
 
+	if cfg.mqttTargetHealthCheck != "" {
+		for {
```

Comment:
should this be in the go routine?

Reply:
Makes sense, good candidate for goroutine.

Reply:
But wait - this is done only on init, and mProxy is blocked here until broker is up. However, I agree that we can continue to check health constantly in the background and restart mProxy if VerneMQ is down.

Reply:
I took a quick look [this](https://medium.com/appsflyer/health-checks-like-a-pro-3fa362942883#id_token=eyJhbGciOiJSUzI1NiIsImtpZCI6ImUxOTdiZjJlODdiZDE5MDU1NzVmOWI2ZTVlYjQyNmVkYTVkNTc0ZTMiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJuYmYiOjE2MDgxNDk1NjQsImF1ZCI6IjIxNjI5NjAzNTgzNC1rMWs2cWUwNjBzMnRwMmEyamFtNGxqZGNtczAwc3R0Zy5hcHBzLmdvb2dsZXVzZXJjb250ZW50LmNvbSIsInN1YiI6IjEwNzQ4NTcwMTQzNTA2MDQ4NDk5NCIsImVtYWlsIjoiZHJhc2tvLmRyYXNrb3ZpY0BnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXpwIjoiMjE2Mjk2MDM1ODM0LWsxazZxZTA2MHMydHAyYTJqYW00bGpkY21zMDBzdHRnLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwibmFtZSI6IkRyYXNrbyBEUkFTS09WSUMiLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDMuZ29vZ2xldXNlcmNvbnRlbnQuY29tL2EtL0FPaDE0R2dYWXBVZ2JLWDRyaXhKQXBwWTBNZGVwSTRROVZoOGJlS1JPdDMxeGc9czk2LWMiLCJnaXZlbl9uYW1lIjoiRHJhc2tvIiwiZmFtaWx5X25hbWUiOiJEUkFTS09WSUMiLCJpYXQiOjE2MDgxNDk4NjQsImV4cCI6MTYwODE1MzQ2NCwianRpIjoiMGJkZGVhZGQyMzVkZmY0MWYyNTk1MzgwOTZhY2RiMTIxODY3MjRhNyJ9.kcO5gj10CVXWR9WKvfAGAVVv-koh_dtEirN08t0t-zeLd6PXn9BGXUaPm6tNPxOBR6QMU_axfi06kTY-8q4utRUE1t6-wfM_TUQge5b6dGv-zC8NfJ2HHXLEVrYI7XQ8hYjvvsk-FQxalogrbvDqFVsPiVVeX8B9wPcZUIz8fadUfFi9H0hFnolul_sOZKG-SqfnMsVeBA3bJ6s7cUZVzdYKW7e1IyYXMGKT9yo7ehSZNIhDMrCoS8oEN9Z-Zgez9qJZJsp0zt3o5OKTGGHDUhVqSzjwZK60tdy_HcelpEnYcr9vMTkTXcESqbiq_oMkoyWbpbM7iWCDPU587wOqpQ) approach seems interesting.

The mentioned [go-sundheit](https://github.com/AppsFlyer/go-sundheit) library allows you to [schedule](https://github.com/AppsFlyer/go-sundheit/blob/c78d43b4328a771cf4c94d8c66e5d16e6c46ad85/health.go#L142) an async check in a [go routine](https://github.com/AppsFlyer/go-sundheit/blob/c78d43b4328a771cf4c94d8c66e5d16e6c46ad85/health.go#L191).

Reply:
Restarting mProxy when VerneMQ is down will be done by kubernetes. There is [PR](https://github.com/mainflux/devops/pull/74) in devops repo that defines liveness probe for mProxy container with checking health checker on VerneMQ. If it's not healthy, both containers (VerneMQ and mProxy) will be restarted by kubernetes. 
It would be redundant and probably conflicting to have same same job done here in code and by orchestration tool.